src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -266,14 +264,19 @@ public async Task ProcessInputAsync(Message message, Flow flow, CancellationToke
             }
             catch (Exception ex)
             {
-                _logger.Error(ex, "Error processing input '{Input}' for user '{User}'", message.Content, userIdentity);
-
                 if (inputTrace != null)
                 {
                     inputTrace.Error = ex.ToString();
                 }
-                
-                throw;
+
+                var builderException = ex is BuilderException be ? be :
+                    new BuilderException($"Error processing input '{message.Content}' for user '{userIdentity}' in state '{state?.Id}'", ex);
```

Comment:
The state processing time is useful too.
Can you please add a property for it on BuilderException and record the stateStopWatch.TotalMilliseconds on it?